pymeasure/instruments/keysight/keysightDSOX1102G.py

```diff
@@ -0,0 +1,647 @@
+#
+# This file is part of the PyMeasure package.
+#
+# Copyright (c) 2013-2020 PyMeasure Developers
+#
+# Permission is hereby granted, free of charge, to any person obtaining a copy
+# of this software and associated documentation files (the "Software"), to deal
+# in the Software without restriction, including without limitation the rights
+# to use, copy, modify, merge, publish, distribute, sublicense, and/or sell
+# copies of the Software, and to permit persons to whom the Software is
+# furnished to do so, subject to the following conditions:
+#
+# The above copyright notice and this permission notice shall be included in
+# all copies or substantial portions of the Software.
+#
+# THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+# IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+# FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+# AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+# LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+# OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN
+# THE SOFTWARE.
+#
+
+import logging
+
+log = logging.getLogger(__name__)
+log.addHandler(logging.NullHandler())
+
+import numpy as np
+from pymeasure.instruments import Instrument
+from pymeasure.instruments.validators import strict_discrete_set, strict_range
+
+
+class KeysightDSOX1102G(Instrument):
+    """ Represents the Keysight DSOX1102G Oscilloscope interface for interacting
+    with the instrument.
+
+    Refer to the Keysight DSOX1102G Oscilloscope Programmer's Guide for further details about
+    using the lower-level methods to interact directly with the scope.
+
+    .. code-block:: python
+        scope = KeysightDSOX1102G(resource)
+        scope.autoscale()
+        ch1_data_array, ch1_preamble = scope.download_data(source="channel1", points=2000)
+        # ...
+        scope.shutdown()
+
+     Known issues:
+        - The digitize command will be completed before the operation is. May lead to
+            VI_ERROR_TMO (timeout) occuring when sending commands immediately after digitize.
+            Current fix: if deemed necessary, add delay between digitize and follow-up command
+            to scope.
+
+    """
+
+    BOOLS = {True: 1, False: 0}
+
+    #################
+    # Channel setup #
+    #################
+
+    def autoscale(self):
+        """ Autoscale displayed channels. """
+        self.write(":autoscale")
+
+    #############
+    # Channel 1 #
+    #############
+
+    @property
+    def ch1(self):
```

Comment:
There seems to be a lot of duplicated code for the two channels. Can you take a look at how another instrument has [implemented channels](https://github.com/ralph-group/pymeasure/blob/07cf309b8d113931525c8cc216cc704f2b7ff413/pymeasure/instruments/tektronix/afg3152c.py#L28) and [their use](https://github.com/ralph-group/pymeasure/blob/07cf309b8d113931525c8cc216cc704f2b7ff413/pymeasure/instruments/tektronix/afg3152c.py#L186-L187) for one pattern how to approach this. Would that be more efficient/elegant in your case, and reduce duplication?